### PR TITLE
Change GitDist to FileDist

### LINKED ISSUE #41: Convert git distributor class to file archive class
The ingestor tool will not deal with git at all. Instead the received XML file will be archived based on its UUID. Since the raw XML data is received, extracting the ID is trivial.

### PR DESCRIPTION
**Pull request by vkbo**
**Summary**:

This PR renames the git class and tests to instead be about archiving files. The archive function will now copy the file into a specified archive folder by generating three levels of up to 16 folders, and save the file as an UUID.xml file under that structure.

**Related issue**:

Closes #41 

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: dmci/distributors/file_dist.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            return False
+
+        mdID = None
+        if self._worker is not None:
+            mdID = self._worker._file_metadata_id
+        if mdID is None:
+            logger.error("No metadata_identifier provided, cannot archive file")
+            return False
+
+        try:
```
Comment (mortenwh) [inline dmci/distributors/file_dist.py:70]: This test should be moved to the worker class

**Review by mortenwh — CHANGES_REQUESTED** [review]
La inn kommentar, så vi husker hva vi skulle gjøre her :)

File: dmci/distributors/file_dist.py (hunk 1 above)
Comment (vkbo) [reply dmci/distributors/file_dist.py:70]: Hmm, the actual check is already done in the worker class. This one is just a safeguard. The return False here also goes back to the API, which should return an error to the user.

**Review by mortenwh — APPROVED** [review]
Looks good! Will you merge @vkbo ?

File: dmci/distributors/file_dist.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        """Add the xml file to the archive.
+        """
+        jobsDir = self._conf.file_archive_path
+        if jobsDir is None:
+            logger.error("No 'file_archive_path' set")
+            return False
+
+        fileUUID = None
+        if self._worker is not None:
+            fileUUID = self._worker._file_metadata_id
```
Comment (johtoblan) [inline dmci/distributors/file_dist.py:65]: What about missing worker, should exit?

File: tests/test_dist/test_file_dist.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    assert tstDist.run() is False
+
+    # Invalid identifier set
+    tstDist._worker = tstWorker
+    goodUUID = tstWorker._file_metadata_id
+    tstWorker._file_metadata_id = "123456789abcdefghijkl"
+    assert tstDist.run() is False
+
+    # Should have a valid identifier from here on
+    tstWorker._file_metadata_id = goodUUID
```
[the remark this answers is not in the document]
Comment (johtoblan) [reply tests/test_dist/test_file_dist.py:101]: The one below!!!

**Review by johtoblan — CHANGES_REQUESTED** [review]
See comments

**Review by johtoblan — APPROVED** [review]

**Pull request merged**